### PR TITLE
Caulaflower give ammo2

### PR DESCRIPTION
**Pull request by Caulaflower**
## Additions

Ammo giving PR. Remade #1416 without mortar stuff

### CODE REVIEW

**Review by perkinslr — DISMISSED** [review]
Unless I'm missing something, this lets the player force ammo into wildlife.  As neat as it might be to stick a couple antigrains onto a boomrat and turn it loose...  We should probably check if the target can use the ammo if it isn't a player pawn.

**Comment by perkinslr** [comment]
If that is more work than you want to do at the moment, I'm happy to approve this for merging, and then follow it up with the needed checks, since I can't directly edit the source branch.  I could also do it as a PR against your source branch if you want.

**Review by ItsSamGray — APPROVED** [review]
LGTM, gotta add the comp to baseline human in this PR or the next one which will be opended for the XML implementation of the system.

**Review by ItsSamGray — APPROVED** [review]

File: Source/CombatExtended/CombatExtended/Comps/CompAmmoGiver.cs [hunk 1]
```diff
@@ -10,6 +10,31 @@
 
 namespace CombatExtended
 {
+    public class Building_WorkTableSmoking : Building_WorkTable
```
Comment (perkinslr) [inline Source/CombatExtended/CombatExtended/Comps/CompAmmoGiver.cs:13]: This does not look related to this PR.

**Review by perkinslr — CHANGES_REQUESTED** [review]

**Review by perkinslr — APPROVED** [review]

**Review by ItsSamGray — APPROVED** [review]

**Pull request merged**